pkg/apis/core/validation/validation.go

```diff
@@ -1770,20 +1770,30 @@ func ValidatePersistentVolume(pv *core.PersistentVolume) field.ErrorList {
 // ValidatePersistentVolumeUpdate tests to see if the update is legal for an end user to make.
 // newPv is updated with fields that cannot be changed.
 func ValidatePersistentVolumeUpdate(newPv, oldPv *core.PersistentVolume) field.ErrorList {
+	newPvClone := newPv.DeepCopy()
```

Comment:
we only need to clone newPv if we're going to nil out ControllerExpandSecretRef. move the clone and field clearing down below and make it conditional:

```go
if oldPv.Spec.CSI != nil && oldPv.Spec.CSI.ControllerExpandSecretRef == nil &&
   newPv.Spec.CSI != nil && newPv.Spec.CSI.ControllerExpandSecretRef != nil {
  // copy and nil ControllerExpandSecretRef to allow the immutable check to pass
  newPv = newPv.DeepCopy()
  newPv.Spec.CSI.ControllerExpandSecretRef = nil
}
```